pkg/controller/account/byoc.go

```diff
@@ -47,16 +47,20 @@ func (r *ReconcileAccount) initializeNewCCSAccount(reqLogger logr.Logger, accoun
 	if acctClaimErr != nil {
 		// TODO: Unrecoverable
 		// TODO: set helpful error message
-		utils.SetAccountClaimStatus(
-			accountClaim,
-			"Failed to get AccountClaim for CSS account",
-			"FailedRetrievingAccountClaim",
-			awsv1alpha1.ClientError,
-			awsv1alpha1.ClaimStatusError,
-		)
-		err := r.Client.Status().Update(context.TODO(), accountClaim)
-		if err != nil {
-			reqLogger.Error(err, "failed to update accountclaim status")
+		if accountClaim != nil {
+			utils.SetAccountClaimStatus(
```

Comment:
Not sure if it's the golang way for param validation but you could potentially move this `accountClaim != nil` check into the `SetAccountClaimStatus` function, as well,  to resolve this for other calls to this func

Reply:
Ahh, that's a great idea!

Reply:
Doesn't look like we have a logger there and I don't feel like adding _additional_ fields to that 🙄 so we'll just exit out for now.